core/src/main/java/org/sonarsource/sonarlint/core/repository/rules/RulesRepository.java

```diff
@@ -26,45 +26,62 @@
 import java.util.stream.Collectors;
 import javax.annotation.CheckForNull;
 import org.sonarsource.sonarlint.core.rule.extractor.SonarLintRuleDefinition;
+import org.sonarsource.sonarlint.core.rules.RulesExtractionHelper;
 
 public class RulesRepository {
+
+  private final RulesExtractionHelper extractionHelper;
   private Map<String, SonarLintRuleDefinition> embeddedRulesByKey;
   private final Map<String, Map<String, SonarLintRuleDefinition>> rulesByKeyByConnectionId = new HashMap<>();
   private final Map<String, Map<String, String>> ruleKeyReplacementsByConnectionId = new HashMap<>();
 
-  public void setEmbeddedRules(Collection<SonarLintRuleDefinition> embeddedRules) {
-    this.embeddedRulesByKey = byKey(embeddedRules);
+  public RulesRepository(RulesExtractionHelper extractionHelper) {
+    this.extractionHelper = extractionHelper;
```

Comment:
I think the original idea behind repositories is that they wouldn't have logic, they would only cache things, but maybe I imposed this rule myself implicitly. Here we rely on another component and do some lazy loading, which is a bit of logic. Not sure it's bad, just that it's a bit different from what we did so far and what I had in mind